src/librustc_typeck/check/method/probe.rs

```diff
@@ -908,8 +957,26 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
     fn pick_core(&mut self) -> Option<PickResult<'tcx>> {
         let steps = self.steps.clone();
 
-        // find the first step that works
-        steps.iter().filter_map(|step| self.pick_step(step)).next()
+        match self.looking_for {
```

Comment:
> I kept the original logic so if it wasn't implemented before, I didn't add it

I don't understand what you mean by "the original logic" here...?

In any case, I agree that this PR doesn't have to properly handle the shadowing case necessarily